ethexe/service/src/lib.rs

```diff
@@ -723,3 +741,61 @@ impl Service {
         Ok(tx_hash)
     }
 }
+
+mod identity {
+    use super::*;
+
+    pub struct ServiceIdentity {
+        validator: Option<Address>,
+        block_duration: u64,
+        block: Option<SimpleBlockData>,
+        producer: Option<Address>,
+    }
+
+    impl ServiceIdentity {
+        pub fn new(validator: Option<Address>, block_duration: u64) -> Self {
+            Self {
+                validator,
+                block_duration,
+                block: None,
+                producer: None,
+            }
+        }
+
+        pub fn receive_new_chain_head(&mut self, block: SimpleBlockData) {
+            self.block = Some(block);
+            self.producer = None;
+        }
+
+        /// Returns whether synced block is previously received chain head.
+        pub fn receive_synced_block(&mut self, data: &BlockSyncedData) -> Result<bool> {
```

Comment:
shouldn't this be performed automatically on receive new head?

Reply:
how to do this without validators information? we can make optimisation after where new validators info could be propagated from previous block validators in most cases